searchlib/src/vespa/searchlib/common/bitvector.cpp

```diff
@@ -199,23 +208,36 @@ BitVector::repairEnds()
 void
 BitVector::andWith(const BitVector & right)
 {
-    verifyContains(*this, right);
+    verifyInclusiveStart(*this, right);
 
-    IAccelrated::getAccelrator().andBit(getActiveStart(), right.getWordIndex(getStartIndex()), getActiveBytes());
+    uint32_t commonBytes = std::min(getActiveBytes(), numActiveBytes(getStartIndex(), right.size()));
+    IAccelrated::getAccelrator().andBit(getActiveStart(), right.getWordIndex(getStartIndex()), commonBytes);
+    if (right.size() < size()) {
+        clearInterval(right.size(), size());
+    }
 
-    setGuardBit();
+    repairEnds();
     invalidateCachedCount();
 }
 
 
 void
 BitVector::andNotWith(const BitVector& right)
 {
-    verifyContains(*this, right);
+    verifyInclusiveStart(*this, right);
 
-    IAccelrated::getAccelrator().andNotBit(getActiveStart(), right.getWordIndex(getStartIndex()), getActiveBytes());
+    if (right.size() < size()) {
+        ssize_t commonBytes = numActiveBytes(getStartIndex(), right.size()) - sizeof(Word);
+        if (commonBytes > 0) {
+            IAccelrated::getAccelrator().andNotBit(getActiveStart(), right.getWordIndex(getStartIndex()), commonBytes);
+        }
+        Index last(right.size() - 1);
```

Comment:
Wraparound when right.size() is 0, followed by bad memory access.

Reply:
Guard added